frontend/types/campaigns.ts

```diff
@@ -36,3 +33,63 @@ export type CampaignSearchFormData = {
   endDateFrom: Nullable<Moment>;
   endDateTo: Nullable<Moment>;
 };
+
+export type CampaignListQueryData = {
+  name?: string;
+  status: {
+    isActive: boolean;
+    isUpcoming: boolean;
+    isCompleted: boolean;
+  };
+  start: {
+    from?: string;
+    to?: string;
+  };
+  end: {
+    from?: string;
+    to?: string;
+  };
```

Comment:
Should the `status`, `start` and `end` properties be possibly undefined as well?

Also, just wanted to ask if this is just an intermediate type before it is converted to the query string.

Reply:
Was worried about the query params having nested objects.

<img width="365" alt="image" src="https://user-images.githubusercontent.com/75519136/195356572-1eb84c23-c06f-4a1d-ae5c-e73c8a2041fa.png">

Seems like there is a way for the nested object to be encoded, not sure how Rails will process it. But I'll try it out since I'm doing campaigns list as well.